### DISCUSSION TITLE
How to put tables in landscape in docx

### DISCUSSION THREAD

**Discussion by lorenzoFabbri** [quarto-dev/quarto-cli#8620] [Q&A]
### Description

I have a very long table that I'd like to *rotate* so that it's in landscape mode. The output is a `docx`. I am using Quarto version 1.4. I am currently experimenting with [this](https://stackoverflow.com/questions/73784720/changing-page-orientation-in-word-using-quarto) `lua` filter.

````qmd
---
format:
  docx:
    citeproc: true
link-citations: true
linkcolor: blue
editor_options: 
  chunk_output_type: console
execute:
  echo: false
  warning: false
  error: false
  message: false
filters:
  - land.lua
---

::: landscape
```{r}
mtcars |>
  gt::gt() |>
  gt::tab_options(
    page.orientation = "landscape"
  )
```
:::
````

It kind of works but I was wondering if there is now a built-in solution to achieve the same. Thank you.

**Comment by mcanouil** [comment]
No built-in, because this is limited to Word documents whilst Quarto can output to more than that.
The Lua filter could be updated to include LaTeX and Typst then be shared as a Quarto extension (<https://quarto.org/docs/extensions/creating.html>).

Related:

- https://github.com/quarto-dev/quarto-cli/discussions/6104
- https://github.com/quarto-dev/quarto-cli/discussions/680
- https://github.com/quarto-dev/quarto-cli/issues/8512

**Reply by lorenzoFabbri** [reply]
Okay, it's a bummer. I am going to say something that's most likely very stupid... But I do not understand what the problem is here. Can't I simply *transpose* the table so that the columns are horizontal rather than vertical?

**Reply by mcanouil** [reply]
To clarify, I did not say it's not going to happen.
I am saying that Quarto wants similar behaviour across formats.
Adding new "asymmetrical" features does not serve that purpose.
There are already some features that need to be translated to other formats and only a limited number of time and people to do that.

The extension mechanism is there exactly to allow users to fill the gap according to their goals.

> But I do not understand what the problem is here. Can't I simply transpose the table so that the columns are horizontal rather than vertical

What does this have to do with landscape/portrait page direction?
You want Quarto to somehow detect from a user generated output that the page (for formats that use pages) to change from portrait to landscape or the other way around automatically?
